base/broadcast.jl

```diff
@@ -1244,22 +1225,21 @@ macro __dot__(x)
     esc(__dot__(x))
 end
 
-function Base.show(io::IO, bc::Broadcasted)
-    print(io, "Broadcasted(", bc.f)
-    args = bc.args
-    while args != TupleLLEnd()
-        print(io, ", ", args.head)
-        args = args.rest
-    end
-    print(io, ')')
-end
+Base.show(io::IO, bc::Broadcasted) = print(io, "Broadcasted(", bc.f, ", ", bc.args, ')')
 
-function make_kwsyntax(f, args...; kwargs...)
-    args′ = make_TupleLL(args...)
-    g = (args...)->f(args...; kwargs...)
-    return Broadcasted(g, args′)
+struct TypeArgFunction{F, T, n} <: Function
+    f::F
 end
-make(f, args...) = make(combine_styles(args...), f, args...)
-make(::S, f, args...) where S<:BroadcastStyle = Broadcasted{S}(f, make_TupleLL(args...))
+(f::TypeArgFunction{F, T, 1})(args...) where {F,T} = f.f(T, args...)
+(f::TypeArgFunction{F, T, 2})(arg1, args...) where {F,T} = f.f(arg1, T, args...)
```

Comment:
I really think limited cases here are ok.  In order for `f` to do something type-stable with the type argument, it must have an explicit method that accepts a `::Type{T}` in a given slot.  We don't need to do the whole-hog "at-least-one-argument-is-a-type" thing, just getting the first few args is sufficient.

Reply:
Seems reasonable to me. Anyone who wants to extend this can write more `_make` methods.